lib/mongoid/relations/eager.rb

```diff
@@ -34,11 +34,18 @@ def eager_load(docs)
       end
 
       def preload(relations, docs)
-
-        relations.group_by do |metadata|
-          metadata.relation
-        end.each do |relation, associations|
-          relation.eager_load_klass.new(associations, docs).run
+        rs = relations.group_by do |metadata|
```

Comment:
Whenever I see a variable `rs` I keep thinking "replica set" or "record set". Better to call this `_relations` IMHO (or `grouped_relations`)